### DISCUSSION TITLE
Is initstmt supported?

### DISCUSSION THREAD

**Discussion by tsaiggo** [mysql-net/MySqlConnector#1370] [Q&A]
My customer is using the properties **initstmt** in the MySQL ODBC Driver.
I am trying to migrate from MySQL ODBC Driver to net/MySqlConnector. But I couldn't find any properties related to **initstmt** in the MySqlConnectionStringBuilder.
Do you have any plans to implement initstmt? Or is there any reason not to implement this property?

This is an explanation related to initstmt
https://dev.mysql.com/doc/connector-odbc/en/connector-odbc-configuration-connection-parameters.html

**Comment by bgrainger** [comment]
There are no plans to support `initstmt`.

A workaround you could try is attaching an event handler to the [`MySqlConnection.StateChange`](https://learn.microsoft.com/en-us/dotnet/api/system.data.common.dbconnection.statechange) event. When `StateChangeEventArgs.CurrentState` changes to `ConnectionState.Open`, execute the SQL you would set as the `initstmt`.
